base/collection_xattr_common.go

```diff
@@ -201,7 +201,7 @@ func WriteUpdateWithXattr(ctx context.Context, store *Collection, k string, xatt
 		}
 
 		// Invoke callback to get updated value
-		updatedValue, updatedXattrValue, isDelete, callbackExpiry, err := callback(value, xattrValue, userXattrValue, cas)
+		updatedValue, updatedXattrValue, isDelete, callbackExpiry, updatedOpts, err := callback(value, xattrValue, userXattrValue, opts, cas)
```

Comment:
I'm on the fence about whether the callback should take opts as an input (as opposed to just having it return []sgbucket.MacroExpansionSpec that would get appended by the caller).  Is there a use case (now or future) where you think the callback might want to reference existing opts content?

Reply:
I was also on the fence about returning the macro expansion spec list or the whole mutate in options. I originally went for the mutate in spec to be returned then changed to return the whole options struct because I felt like it gave us more freedom to expand upon in future if needed. But looking at it this morning after your comments I agree, i'll update the PRs to reflect this.